crypto/storage/client.go

```diff
@@ -0,0 +1,109 @@
+package storage
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/crypto/storage/httpclient"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+	"net/http"
+)
+
+// APIClient implements the Storage interface. It uses a simple HTTP protocol to connect to an external storage server.
+// This server can either be a secret store itself, or proxy the request to a key store such as Hashicorp Vault or Azure Key Vault.
+// It allows us to keep the codebase clean and allow other parties to write their own adaptor.
+type APIClient struct {
+	httpClient *httpclient.ClientWithResponses
+}
+
+type APIClientConfig struct {
+	URL string `koanf:"url"`
+}
+
+func NewAPIClient(url string) (Storage, error) {
+	client, _ := httpclient.NewClientWithResponses(url)
+	response, err := client.HealthCheckWithResponse(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to storage server: %w", err)
+	}
+	if response.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unable to connect to storage server: unexpected status code: %d", response.StatusCode())
```

Comment:
not sure whether this is a good idea, in cloud environment containers should be able to start even if it depends on something that's not (yet) running (might still be starting).

Better to have the health check of Nuts Node execute this health check (although be wary of slow responses, so configure time-outs tightly), and have the environment deal with it (monitoring/restarting/etc)?

Reply:
I've implemented the `HealthCheckable` interface, but this currently only results in the `/health` endpoint in reporting to be down. The node still continues to run. Keys will get lost if a LB doesn't catch this.